src/beamlime/applications/daemons.py

```diff
@@ -49,45 +96,25 @@ def __init__(
         frame_rate: FrameRate,
     ):
         self.logger = logger
-        self.nexus_container = NexusContainer.from_template_file(nexus_template_path)
-        event_rate_per_detector = int(
-            event_rate / max(len(self.nexus_container.detectors), 1)
-        )
-        detector_names = sorted(self.nexus_container.detectors.keys())
-        ev44_source_names = self._collect_detector_ev44_modules()
-
-        self.random_event_generators = {
-            det_name: random_ev44_generator(
-                source_name=DetectorName(ev44_source_name),
-                detector_numbers=DetectorNumberCandidates(
-                    self.nexus_container.detectors[det_name].pixel_ids
-                ),
-                event_rate=EventRate(event_rate_per_detector),
-                frame_rate=frame_rate,
-            )
-            for det_name, ev44_source_name in zip(detector_names, ev44_source_names)
-            # Assuming the order of the detectors and the ev44 modules are the same
-            # For real data stream, it doesn't matter
-            # since the data is inserted by the source name.
-        }
-        self.data_feeding_speed = speed
-        self.num_frames = num_frames
 
-    def _collect_detector_ev44_modules(self) -> list[str]:
-        """Assumig 'detector' is included in the topic of the module."""
+        with open(nexus_template_path) as f:
+            self.nexus_structure = json.load(f)
+
+        if self.nexus_structure['children'][0]['children'][-1] == '$USERS$':
```

Comment:
Aha, this should be removed.
The nexus structure templates in gitlab have some templating variables that I assume will not be present in the run-start messages. The presence of the templating variables messes up the parsing of the nexus structure, thats why this was needed here to make the tests pass.

I think the better option would be to remove the templating variable directly in the json file and store that version in putch.